core/trino-main/src/test/java/io/trino/block/BlockAssertions.java

```diff
@@ -596,4 +626,237 @@ public static RunLengthEncodedBlock createRLEBlock(long value, int positionCount
+    {
+        return createStringsBlock(
+                generateListWithNulls(positionCount, nullRate, () -> generateRandomStringWithLength(maxStringLength)));
+    }
+
+    private static String generateRandomStringWithLength(int length)
+    {
+        String symbols = "abcdefghijklmnopqrstuvwxyz";
+        char[] chars = new char[length];
+        for (int i = 0; i < length; i++) {
+            chars[i] = symbols.charAt(RANDOM.nextInt(symbols.length()));
+        }
+        return new String(chars);
+    }
+
+    public static RunLengthEncodedBlock createRleBlockWithRandomValue(Block block, int positionCount)
+    {
+        checkArgument(block.getPositionCount() > 0, format("block positions %d is less than or equal to 0", block.getPositionCount()));
+        return new RunLengthEncodedBlock(block.getRegion(block.getPositionCount() / 2, 1), positionCount);
+    }
+
+    public static Block wrapBlock(Block block, int positionCount, List<Encoding> wrappings)
+    {
+        checkArgument(!wrappings.isEmpty(), "wrappings is empty");
+
+        Block wrappedBlock = block;
+
+        for (int i = wrappings.size() - 1; i >= 0; i--) {
+            switch (wrappings.get(i)) {
+                case DICTIONARY:
+                    wrappedBlock = createRandomDictionaryBlock(wrappedBlock, positionCount, true);
+                    break;
+                case RUN_LENGTH:
+                    wrappedBlock = createRleBlockWithRandomValue(wrappedBlock, positionCount);
+                    break;
+                default:
+                    throw new IllegalArgumentException(format("wrappings %s is incorrect", wrappings));
+            }
+        }
+        return wrappedBlock;
+    }
+
+    static <T> Iterable<T> generateListWithNulls(int positionCount, float nullRate, Supplier<T> valueSupplier)
+    {
+        List<T> result = new ArrayList<>(positionCount);
+
+        Set<Integer> nullPositions = chooseNullPositions(positionCount, nullRate);
+        for (int i = 0; i < positionCount; i++) {
+            result.add(nullPositions.contains(i) ? null : valueSupplier.get());
+        }
+        return unmodifiableList(result);
+    }
+
+    private static Set<Integer> chooseNullPositions(int positionCount, float nullRate)
+    {
+        int nullCount = (int) (positionCount * nullRate);
+        if (nullCount == 0 && nullRate > 0) {
+            throw new IllegalArgumentException(format("position count %d too log to have at least one null with rate %f", positionCount, nullRate));
+        }
+        Set<Integer> nullPositions = chooseRandomUnique(positionCount, nullCount);
+        return nullPositions;
+    }
+
+    private static Set<Integer> chooseRandomUnique(int bound, int count)
+    {
+        List<Integer> allNumbers = IntStream.range(0, bound).boxed().collect(toList());
+        Collections.shuffle(allNumbers, RANDOM);
+        return allNumbers.stream().limit(count).collect(toUnmodifiableSet());
```

Comment:
We use Guava `ImmutableSets` across code so that `ImmutableSet.copyOf` is optimized